clientconn.go

```diff
@@ -118,6 +118,15 @@ func WithPicker(p Picker) DialOption {
 // WithBackoffConfig configures the dialer to use the provided backoff
 // parameters after connection failures.
 func WithBackoffConfig(b *BackoffConfig) DialOption {
```

Comment:
What do u mean by "partial"? This proposal is equivalent to what you did in this PR.